File: server/routes/genreRoute.js
const express = require("express");
const router = express.Router();
const Streamer = require('../models/Streamer');

 router.post("/genre", async (req, res) => {
  console.log("Received body:", req.body); 

  const { username, genres } = req.body;

  if (!username || !genres) {
    return res.status(400).json({ error: "Missing userId or genres" });
  }

  try {
    const user = await Streamer.findOne({ username });
    if (!user) {
      return res.status(404).json({ error: "User not found" });
    }

    user.genres = genres;
    await user.save();

    res.status(200).json({ message: "Preferences updated successfully" });
  } catch (error) {
    console.error("Error saving preferences:", error);
    res.status(500).json({ error: "Internal server error" });
  }
});


module.exports = router;

File: backend/server/routes/genreRoute.py
from fastapi import APIRouter, Request, HTTPException
from pydantic import BaseModel
from fastapi.responses import JSONResponse

router = APIRouter()

class GenreRequest(BaseModel):
    username: str
    genres: list[str]

@router.post("/preference/genre")
async def update_genres(data: GenreRequest, request: Request):
    db = request.app.state.user_db
    collection = db["streamer"]  # your MongoDB collection is named 'streamer'

    user = collection.find_one({"username": data.username})
    if not user:
        raise HTTPException(status_code=404, detail="User not found")

    try:
        collection.update_one(
            {"username": data.username},
            {"$set": {"genres": data.genres}}
        )
        return JSONResponse(content={"message": "Preferences updated successfully"}, status_code=200)
    except Exception as e:
        print("Error saving preferences:", e)

        raise HTTPException(status_code=500, detail="Internal server error")